app/hotels/src/allHotels/AllHotelsSearch.js

```diff
@@ -14,21 +13,23 @@ import type { Coordinates } from '../CoordinatesType';
 import type { AllHotelsSearchQueryResponse } from './__generated__/AllHotelsSearchQuery.graphql';
 import type { FilterParams } from '../filter/FilterParametersType';
 import type { SearchParams } from './searchForm/SearchParametersType';
-import type { FilterReducerState } from '../filter/FiltersReducer';
-import type { HotelsReducerState } from '../HotelsReducer';
+import HotelsSearchContext from '../HotelsSearchContext';
+import HotelsFilterContext from '../HotelsFilterContext';
 
-type Props = {|
-  search: SearchParams,
+type PropsWithContext = {
+  // external:
   openSingleHotel: (searchParams: AvailableHotelSearchInput) => void,
-  onCityIdChange: (cityId: string | null) => void,
   cityId: string,
   coordinates: Coordinates | null,
+  currency: string,
+  // from context
```

Comment:
Why are you not using `...Props` here as done in the other components?

Reply:
I tried but I had to use non-exact props and for some reason, it doesn't work. Can you please check it what's wrong? I wasn't able to fix it this way... 🙈

Reply:
I will have a look and see if I can find out

Reply:
Ok, I can't really find out why this doesn't work. We could merge it and try to make it exact type later in another PR.

Reply:
Should be possible really soon: https://github.com/facebook/flow/issues/2405#issuecomment-379175609